### PR TITLE
Fixed CI

### PR DESCRIPTION
**Pull request by s1lver**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any

### CODE REVIEW

**Comment by s1lver** [comment]
scrutinizer (

**Comment by s1lver** [comment]
Снимок экрана 2021-11-24 в 15 38 11
Now it works

**Comment by s1lver** [comment]
mutation and static - ok

**Comment by vjik** [comment]
Maybe use `GITHUB_ENV` ([docs](https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#setting-an-environment-variable)) how it is done in package template: https://github.com/yiisoft/package-template/blob/master/.github/workflows/build.yml ?

**Comment by s1lver** [comment]
In static.yml 

> (Consider upgrading from Phan 3.2.10 to Phan 4, which has the latest features and bug fixes. This notice can be disabled with PHAN_SUPPRESS_PHP_UPGRADE_NOTICE=1))

**Comment by s1lver** [comment]
> Maybe use `GITHUB_ENV` ([docs](https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#setting-an-environment-variable)) how it is done in package template: https://github.com/yiisoft/package-template/blob/master/.github/workflows/build.yml ?

can it be like this

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
         "yiisoft/yii-queue": "*"
     },
     "require-dev": {
-        "phan/phan": "^3.0",
+        "phan/phan": "^5.0",
```
Comment (vjik) [inline composer.json:32]: We switched phan to psalm. Can remove `phan/phan` dependency and use in CI psalm.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -53,16 +52,28 @@ jobs:
 
       - name: Determine composer cache directory on Linux
         if: matrix.os == 'ubuntu-latest'
-        run: echo "::set-env name=COMPOSER_CACHE_DIR::$(composer config cache-dir)"
+        id: composer-cache-ubuntu
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
```
Comment (vjik) [inline .github/workflows/build.yml:56]: Use `$GITHUB_ENV`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -53,16 +52,28 @@ jobs:
...
 
       - name: Determine composer cache directory on Windows
         if: matrix.os == 'windows-latest'
-        run: ECHO "::set-env name=COMPOSER_CACHE_DIR::~\AppData\Local\Composer"
+        id: composer-cache-windows
+        run: echo "::set-output name=dir::~\AppData\Local\Composer"
```
Comment (vjik) [inline .github/workflows/build.yml:61]: See https://github.com/yiisoft/package-template/blob/d3a99793d83219481edae91553ce8d1a5e93de0d/.github/workflows/build.yml#L66

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -53,16 +52,28 @@ jobs:
...
+      - name: Cache dependencies installed with composer on Linux
+        if: matrix.os == 'ubuntu-latest'
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache-ubuntu.outputs.dir }}
+          key: php${{ matrix.php-version }}-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: |
+            php${{ matrix.php-version }}-composer-
+
+      - name: Cache dependencies installed with composer on Windows
```
Comment (vjik) [inline .github/workflows/build.yml:72]: We can use one step for linux and windows. See https://github.com/yiisoft/package-template/blob/d3a99793d83219481edae91553ce8d1a5e93de0d/.github/workflows/build.yml#L68

Comment (s1lver) [reply .github/workflows/build.yml]: fix

**Review by vjik — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

**Pull request merged**